akka-remote/src/main/scala/akka/remote/artery/ArterySubsystem.scala

```diff
@@ -39,12 +47,15 @@ private[remote] class ArterySubsystem(_system: ExtendedActorSystem, _provider: R
   override def start(): Unit = {
     // TODO: Configure materializer properly
     // TODO: Have a supervisor actor
-    address = Address("akka.artery", system.name, remoteSettings.ArteryHostname, remoteSettings.ArteryPort)
+    localAddress = UniqueAddress(
+      Address("akka.artery", system.name, remoteSettings.ArteryHostname, remoteSettings.ArteryPort),
+      AddressUidExtension(system).addressUid)
     materializer = ActorMaterializer()(system)
 
     transport =
       new Transport(
-        address,
+        localAddress,
+        this, //FIXME should we collapse ArterySubsystem and Transport into one class?
```

Comment:
@drewhk do you think it makes sense to collapse `ArterySubsystem` and `Transport` to one class? I don't see the need for separating them.